src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -659,6 +667,11 @@ public Solution AddDocument(DocumentId documentId, string name, SourceText text,
 
             var project = _state.GetProjectState(documentId.ProjectId);
 
+            if (project == null)
```

Comment:
This was a legitimate bug that was caught: we would have thrown in the GetSourceCodeKind call so this isn't a breaking change other than it's now throwing a reasonable exception type.